Sources/Networking/Modifiers/Processors/EndpointRequestStorageProcessor/EndpointRequestStorageProcessor.swift

```diff
@@ -19,28 +20,42 @@ import Foundation
 ///
 /// The filename is created from a sessionId and a corresponding request identifier.
 /// Stored files are stored under session folder and can be added to NSAssetCatalog and read via `SampleDataNetworking` to replay whole session.
-open class EndpointRequestStorageProcessor: ResponseProcessing, ErrorProcessing {
+public class EndpointRequestStorageProcessor: ResponseProcessing, ErrorProcessing {
     // MARK: Private variables
     private let fileManager: FileManager
     private let jsonEncoder: JSONEncoder
     private let config: Config
-    
     private lazy var responsesDirectory = fileManager.temporaryDirectory.appendingPathComponent("responses")
     private lazy var requestCounter = Counter()
-    private lazy var multipeerConnectivityManager: MultipeerConnectivityManager? = {
-        #if DEBUG
-        // Initialise only in DEBUG mode otherwise it could pose a security risk for production apps.
-        guard let multiPeerSharingConfig = config.multiPeerSharing else {
+
+    // This would ideally also be a lazy var, however it has to be async, because UIDevice.current.name needs to be called on MainActor.
+    private var _multipeerConnectivityManager: MultipeerConnectivityManager?
+    private var multipeerConnectivityManager: MultipeerConnectivityManager? {
```

Comment:
in case like this, we touch device just for name. I'm thinking maybe the whole design of networking is bad in this case. For this variable maybe we should inject name to the processor and we don't have to figure out where is it from (app will send it - either device name or custom value)

Reply:
I know, It's a weird implementation, but otherwise it won't be possible to have a shared instance, because the `static let shared` cannot execute the MainActor isolated code `UIDevice.current.name`.